[add] discuss: specific email server from (#2627)
odoo/documentation: content/applications/general/email_communication/email_servers.rst

   :menuselection:`Settings --> Technical --> Automation --> Scheduled Actions` and look for
   :guilabel:`Mail: Fetchmail Service`.

Set up different outgoing email servers for a multi-company environment
=======================================================================

The "From Filter" allows for the use of a specific outgoing email server depending on the "From"
email address that Odoo is sending on behalf of. When an email is sent from Odoo and the "From

> Filter" is configured, then the email server will be chosen as follows:

Reviewer: Thanks for the suggestions @StraubCreative, I've made the changes and added a screenshot after line 196.

#. Odoo searches for an email server having the same "From Filter" as the "From" email address
   defined in the outgoing email. For example, if the "From" email address is `test\@example.com`,
   only the email servers that have the "From Filter" equal to `test\@example.com` will be returned.
#. If no email servers are found, then Odoo searches for an email server that has the same domain
   in the "From Filter" as the "From" email address of the outgoing email. For example, if the
   "From" email address is `test\@example.com`, only the email servers that have the "From Filter"

>    equal to `example.com` will be returned.

Reviewer: Made these changes, thanks for editing. :)

> If no email servers are found after checking for the domain, then Odoo will return all email
> servers that do not have any "From Filter" set.

Reviewer 2: `:guilabel:`
```suggestion
If no email servers are found after checking for the domain, then Odoo will return all email
servers that do not have any "FROM Filtering" values set.
```

> Should this previous query return no results, then Odoo performs a search for an email server
> using the system parameter: `mail.default.from`. First, the email address listed will attempt
> to match an email server and then the domain will attempt a match.
> 
> If no email server is found then Odoo will return the first outgoing email server (sorted by
> priority).

Reviewer 2: Line broke too early on line 214.

As well, here's an optional suggestion:
Would connect that last paragraph to the main body preceding it since it's the last step in a sequence of operations... it seemed odd that paragraph would be on it's own.

```suggestion
Should this previous query return no results, then Odoo performs a search for an email server using
the system parameter: `mail.default.from`. First, the email address listed will attempt to match an
email server and then the domain will attempt a match. If no email server is found then Odoo will
return the first outgoing email server (sorted by priority).
```

   :menuselection:`Settings --> Technical --> Automation --> Scheduled Actions` and look for
   :guilabel:`Mail: Fetchmail Service`.

Set up different outgoing email servers for a multi-company environment
=======================================================================

The :guilabel:`FROM Filtering` field allows for the use of a specific outgoing email server

> depending on the :guilabel:`From` email address that Odoo is sending on behalf of. Access this

Reviewer 3: After the first sentence (`The...on behalf of`), I would add a real-life example (1-3 sentences) of why a company might want to use this feature. The section header says it's for multi-company environments. Try to simplify for the reader and help them visualize how they would apply this section's knowledge to their business.

Reviewer: @mivu-odoo I've added two separate sentences: 1. to improve deliverability  2. In a multi-company environment

The title seemed a bit too specific and I've made it more general as multi-company sending is not the only application. I've changed it to: `Utilizing the FROM Filter on an outgoing email server`

> field by navigating to :menuselection:`Settings --> Custom Mail Servers --> Outgoing Mail Servers
> --> New`.

Reviewer 3: ```suggestion
field in Odoo by navigating to :menuselection:`Settings --> Custom Mail Servers --> Outgoing Mail
Servers --> New`.
```

>    .. image:: email_servers/from-filter-setting.png
>       :align: center
>       :alt: Outgoing email server settings and the FROM filter settings.

Reviewer 3: ```suggestion
.. image:: email_servers/from-filter-setting.png
   :align: center
   :alt: Outgoing email server settings and the FROM filter settings.
```

When an email is sent from Odoo while the :guilabel:`FROM Filtering` field is set, an

> email server will then be chosen in the following sequence:

Reviewer 3: ```suggestion
email server is chosen in the following sequence:
```

> #. First, Odoo searches for an email server having the same :guilabel:`FROM Filtering` value as the
>    :guilabel:`From` email address defined in the outgoing email. For example, if the
>    :guilabel:`From` email address is `test\@example.com`, only the email servers that have the
>    :guilabel:`FROM Filtering` value equal to `test\@example.com` will be returned.
> #. However, if no email servers are found using the :guilabel:`From` value, then Odoo searches for
>    an email server that has the same domain in the :guilabel:`FROM Filtering` field as the
>    :guilabel:`From` email address of the outgoing email. For example, if the :guilabel:`From`
>    email address is `test\@example.com`, only the email servers that have the :guilabel:`FROM
>    Filtering` value equal to `example.com` will be returned.

Reviewer 3: ```suggestion
#. First, Odoo searches for an email server that has the same :guilabel:`FROM Filtering` value as
   the :guilabel:`From` value (email address) defined in the outgoing email. For example, if the
   :guilabel:`From` value (email address) is `test\@example.com`, only the email servers that have
   the :guilabel:`FROM Filtering` value equal to `test\@example.com` are returned.
#. However, if no email servers are found that use the :guilabel:`From` value, then Odoo searches
   for an email server that has the same *domain* as the :guilabel:`From` value (email address)
   defined in the outgoing email. For example, if the :guilabel:`From` email address is
   `test\@example.com`, only the email servers that have the :guilabel:`FROM Filtering` value equal
   to `example.com` are returned.
```

> If no email servers are found after checking for the domain, then Odoo will return all email
> servers that do not have any :guilabel:`FROM Filtering` value(s) set.

Reviewer 3: ```suggestion
If no email servers are found after checking for the domain, then Odoo returns all email servers
that do not have any :guilabel:`FROM Filtering` value(s) set.
```

> Should this previous query return no results, then Odoo performs a search for an email server using

Reviewer 3: Clarify `this previous query` - which one are we talking about?

> the system parameter: `mail.default.from`. First, the email address listed will attempt to match an
> email server and then the domain will attempt a match. If no email server is found then Odoo will
> return the first outgoing email server (sorted by priority).

Reviewer 3: ```suggestion
the system parameter: `mail.default.from`. First, the email address listed attempts to match an
email server, and then the domain attempts to find a match. If no email server is found, Odoo
returns the first outgoing email server (sorted by priority).
```

Also clarify `the first outgoing email server` - is it the first one in the list on the `Outgoing Servers` page in Odoo?

.. note::

>    If several email servers are found, then Odoo will use the first one according to its priority.
>    For example, if there are two email servers, one with a priority of `10` and the other with a
>    priority of `20`, then the email server with a priority of `10` will be used.

Reviewer 3: ```suggestion
   If several email servers are found, then Odoo uses the first one according to its priority. For
   example, if there are two email servers, one with a priority of `10` and the other with a
   priority of `20`, then the email server with a priority of `10` is used first.
```

The "From Filter" allows for the use of a specific outgoing email server depending on the "From"
email address that Odoo is sending on behalf of. When an email is sent from Odoo and the "From
Filter" is configured, then the email server will be chosen as follows:

#. Odoo searches for an email server having the same "From Filter" as the "From" email address
   defined in the outgoing email. For example, if the "From" email address is `test\@example.com`,
   only the email servers that have the "From Filter" equal to `test\@example.com` will be returned.

#. If no email servers are found, then Odoo searches for an email server that has the same domain 
   in the "From Filter" as the "From" email address of the outgoing email. For example, if the "From"
   email address is `test\@example.com`, only the email servers that have the "From Filter" equal to
   `example.com` will be returned.

If no email servers are found after checking for the domain, then Odoo will return all email
servers that do not have any "From Filter" set.

Should this previous query return no results, then Odoo performs a search for an email server

> using the system parameter: 'mail.default.from'. First, the email address listed will attempt

Reviewer 3: ```suggestion
using the system parameter: `mail.default.from`. First, the email address listed will attempt
```

   :menuselection:`Settings --> Technical --> Automation --> Scheduled Actions` and look for
   :guilabel:`Mail: Fetchmail Service`.

> Setting up different outgoing email servers for a multi-company environment
> ===========================================================================

Reviewer 3: ```suggestion
Set up different outgoing email servers for a multi-company environment
=======================================================================
```

The "From Filter" allows for the use of a specific outgoing email server depending on the "From"
email address that Odoo is sending on behalf of. When an email is sent from Odoo and the "From
Filter" is configured, then the email server will be chosen as follows:

#. Odoo searches for an email server having the same "From Filter" as the "From" email address

>    defined in the outgoing email. For example, if the "From" email address is test\@example.com,
>    only the email servers having the "From Filter" equal to test\@example.com will be returned.

Reviewer 3: ```suggestion
   defined in the outgoing email. For example, if the "From" email address is `test\@example.com`,
   only the email servers that have the "From Filter" equal to `test\@example.com` will be returned.
```

> #. If no email servers are found, then Odoo searches for an email server having the same domain in
>    the "From Filter" as in the "From" email address of the outgoing email. For example, if the
>    "From" email address is test\@example.com, only the email servers having the "From Filter" equal
>    to example.com will be returned.

Reviewer 3: ```suggestion
#. If no email servers are found, then Odoo searches for an email server that has the same domain in
   the "From Filter" as the "From" email address of the outgoing email. For example, if the "From"
   email address is `test\@example.com`, only the email servers that have the "From Filter" equal to
   `example.com` will be returned.
```

If no email servers are found after checking for the domain, then Odoo will return all email

> servers without any "From Filter" set.

Reviewer 3: ```suggestion
servers that do not have any "From Filter" set.
```

Should this previous query return no results, then Odoo performs a search for an email server
using the system parameter: 'mail.default.from'. First, the email address listed will attempt
to match an email server and then the domain will attempt a match.

If no email server is found then Odoo will return the first outgoing email server (sorted by
priority). In this case, it is likely the email configuration will be spoofing the email and it

> will be flagged as Spam.

Reviewer 3: ```suggestion
will be flagged as **Spam**.
```

.. note::
   If several email servers are found, then Odoo will use the first one according to its priority.

>    For example, if there are two email servers, one with a priority of 10 and the other with a
>    priority of 20, then the email server with a priority of 10 will be used.

Reviewer 3: ```suggestion
   For example, if there are two email servers, one with a priority of `10` and the other with a
   priority of `20`, then the email server with a priority of `10` will be used.
```
